File: kala-game-service/src/main/java/org/jesperancinha/games/kalagameservice/dto/PitDto.java
package org.jesperancinha.games.kalagameservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.jesperancinha.games.kalagameservice.model.Pit;
import org.jesperancinha.games.kalagameservice.model.PitType;

@Setter
@Getter
@Builder
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
public class PitDto {
    private Long id;

    private PitType pitType;

    private Integer stones;

    private PlayerDto playerDto;
}

File: kala-game-service/src/main/kotlin/org/jesperancinha/games/kalagameservice/dto/PitDto.kt
package org.jesperancinha.games.kalagameservice.dto

import com.fasterxml.jackson.annotation.JsonProperty
import org.jesperancinha.games.kalagameservice.model.PitType

data class PitDto(
    @JsonProperty("id")
    val id: Long?,
    @JsonProperty("pitType")
    val pitType: PitType?,
    @JsonProperty("stones")
    val stones: Int?,
    @JsonProperty("playerDto")
    val playerDto: PlayerDto?,
)